Mathlib/RingTheory/IntegralClosure/IsIntegralClosure/Basic.lean

```diff
@@ -48,6 +48,22 @@ theorem isIntegral_algEquiv {A B : Type*} [Ring A] [Ring B] [Algebra R A] [Algeb
     (f : A ≃ₐ[R] B) {x : A} : IsIntegral R (f x) ↔ IsIntegral R x :=
   ⟨fun h ↦ by simpa using h.map f.symm, IsIntegral.map f⟩
 
+/- If `R` and `T` are isomorphic commutative rings and `S` is an `R`-algebra and a `T`-algebra in
+  a compatible way, then an element `a ∈ S` is integral over `R` if and only if it is integral
+  over `T`.-/
+theorem isIntegral_iff_of_equiv {R S T : Type*} [CommRing R] [CommRing S] [CommRing T]
+    [Algebra R S] [Algebra T S] (φ : R ≃+* T)
```

Comment:
Could this be done by promoting `φ` (or `φ.symm`) into an `Algebra` instance, and setting up a `ScalarTower`, and then applying a lemma that says that given `C/B/A` and `c : C`, then `IsIntegral A c -> IsIntegral B c`?

One reason why I'm asking is that I would hope that the `algebraize` tactic (not in mathlib yet, but a prototype was written during the AG workshop a few months ago) would be able to golf the first two steps.